qa/rpc-tests/test_framework/util.py

```diff
@@ -494,3 +494,18 @@ def get_coinbase_address(node, expected_utxos=None):
     addrs = [a for a in set(addrs) if addrs.count(a) == expected_utxos]
     assert(len(addrs) > 0)
     return addrs[0]
+
+def check_node_log(self, node_number, line_to_check, stop_node = True):
+    print("Checking node " + str(node_number) + " logs")
+    if stop_node:
+        self.nodes[node_number].stop()
+        bitcoind_processes[node_number].wait()
+    logpath = self.options.tmpdir + "/node" + str(node_number) + "/regtest/debug.log"
+    foundErrorMsg = False
+    with open(logpath, "r") as myfile:
+        logdata = myfile.readlines()
+    for logline in logdata:
+        if line_to_check in logline:
+            foundErrorMsg = True
+            break
+    assert(foundErrorMsg)
```

Comment:
One of the tests below requires knowing the order in which lines were found.

```suggestion
    with open(logpath, "r") as myfile:
        logdata = myfile.readlines()
    for (n, logline) in enumerate(logdata):
        if line_to_check in logline:
            return n
    raise AssertionError(repr(line_to_check) + " not found")
```